LoadFile.go

```diff
@@ -0,0 +1,81 @@
+package crossrefindexer
+
+import (
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func readFile(path string) error {
+	file, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	gzr, err := GzipReader(file)
+	if err != nil {
+		return err
+	}
+
+	fileType, err := ClassifyDataFormat(gzr)
+	if err != nil {
+		return err
+	}
+
+	if _, err := file.Seek(0, io.SeekStart); err != nil {
+		return err
+	}
+
+	gzr, err = GzipReader(file)
+	if err != nil {
+		return err
+	}
+
+	publications := make(chan CrossRef)
+
+	pubs := []CrossRef{}
+
+	go func() {
+		for {
+			pub, open := <-publications
+			if !open {
+				break
+			}
+			// TODO: Convert to simplified format
+			// TODO: Send on another channel
+			pubs = append(pubs, pub)
+		}
+	}()
```

Comment:
Does this belong with the file reading? I'd say it's better to pass `publications` as an argument

Reply:
I have tried to refactor it but it did not work.
I have tried GPT but its code did not work too, but I kept the code as comment because I felt it could work.
I think this is because of my little understand of the channels